x/farming/keeper/proposal_handler.go

```diff
@@ -225,6 +225,12 @@ func (k Keeper) DeletePublicPlanProposal(ctx sdk.Context, proposals []*types.Del
 			return sdkerrors.Wrapf(sdkerrors.ErrNotFound, "plan %d is not found", p.GetPlanId())
 		}
 
+		if plan.GetFarmingPoolAddress().String() != plan.GetTerminationAddress().String() {
```

Comment:
even if a plan have same `TerminationAddress` with `FarmingPoolAddress` the plan should be terminated with emitting events and setting `plan.terminated` to `true`

so, I think this if statements should be on `TerminatePlan` below line https://github.com/tendermint/farming/blob/783b9420577443d9aa020cb4f781f05382bed918/x/farming/keeper/plan.go#L244